src/coreclr/jit/fgehopt.cpp

```diff
@@ -2461,3 +2461,677 @@ PhaseStatus Compiler::fgTailMergeThrows()
+            //
+            if (inTry)
+            {
+                unsigned const trueAcdTryIndex = add->acdTryIndex - 1;
+
+                if ((trueAcdTryIndex >= originalLowestRegionIndex) && (trueAcdTryIndex <= originalOutermostRegionIndex))
+                {
+                    needsCloningForTry = true;
+                }
+            }
+
+            if (inHnd)
+            {
+                unsigned const trueAcdHndIndex = add->acdHndIndex - 1;
+
+                if ((trueAcdHndIndex >= originalLowestRegionIndex) && (trueAcdHndIndex <= originalOutermostRegionIndex))
+                {
+                    needsCloningForHnd = true;
+                }
+            }
+
+            if (!needsCloningForTry && !needsCloningForHnd)
+            {
+                continue;
+            }
+
+            JITDUMP("Will need to clone: ");
+            JITDUMPEXEC(add->Dump());
+
+            AddCodeDsc* clone = new (this, CMK_Unknown) AddCodeDsc;
+            clone->acdDstBlk  = nullptr;
+
+            if (needsCloningForTry)
+            {
+                clone->acdTryIndex = (unsigned short)(add->acdTryIndex + indexShift);
+            }
+            else if (inTry)
+            {
+                clone->acdTryIndex = add->acdTryIndex;
+            }
+            else
+            {
+                clone->acdTryIndex = 0;
+            }
+
+            if (needsCloningForHnd)
+            {
+                clone->acdHndIndex = (unsigned short)(add->acdHndIndex + indexShift);
+            }
+            else if (inHnd)
+            {
+                clone->acdHndIndex = add->acdHndIndex;
+            }
+            else
+            {
+                clone->acdHndIndex = 0;
+            }
+
+            clone->acdKeyDsg = add->acdKeyDsg;
+            clone->acdKind   = add->acdKind;
+            clone->acdUsed   = false;
+
+#if !FEATURE_FIXED_OUT_ARGS
+            clone->acdStkLvl     = 0;
+            clone->acdStkLvlInit = false;
+#endif // !FEATURE_FIXED_OUT_ARGS
+            INDEBUG(clone->acdNum = acdCount++);
+            cloned.Push(clone);
+        }
+
+        while (cloned.Height() > 0)
+        {
+            AddCodeDsc* const clone = cloned.Pop();
+            AddCodeDscKey     key(clone);
+            map->Set(key, clone);
+            JITDUMP("Added clone: ");
+            JITDUMPEXEC(clone->Dump());
+        }
+    }
+
+    BasicBlock* const clonedTryEntry = (*map)[tryEntry];
+    JITDUMP("Done cloning, cloned try entry is " FMT_BB "\n", clonedTryEntry->bbNum);
+    return clonedTryEntry;
+}
+
+//------------------------------------------------------------------------
+// fgCanCloneTryRegion: see if a try region can be cloned
+//
+// Arguments:
+//    tryEntry - try entry block
+//
+// Returns:
+//    true if try region is clonable
+//
+bool Compiler::fgCanCloneTryRegion(BasicBlock* tryEntry)
+{
+    assert(bbIsTryBeg(tryEntry));
+
+    BitVecTraits      traits(compBasicBlockID, this);
```

Comment:
I would also consider just leaving it up to `fgCloneTryRegion` to create the traits and bit vector of visited blocks to simplify the contract here. It will create another bit vector in the case where we duplicate a loop, but that does not seem like a big deal.

Reply:
> I would also consider just leaving it up to `fgCloneTryRegion` to create the traits and bit vector of visited blocks to simplify the contract here. It will create another bit vector in the case where we duplicate a loop, but that does not seem like a big deal.

Did that, and am now using `bbID` everywhere.